example/extensions/lib_subgraph/README.md

```diff
@@ -0,0 +1,193 @@
+
+```
+hybridize(backend=None, backend_opts=None)
+```
+
+When the `hybridize` function is called, Gluon will convert the program’s execution into the style used in symbolic programming. The `backend` argument is a string that identifies which backend to partition the model for. The `backend_opts` takes other user-specified options that will be passed to the backend partitioning APIs.
+
+### Writing A Custom Partitioner
+
+There are several essential building blocks for making a custom partitioner:
+
+* [initialize](./subgraph_lib.cc#L242):
+    * This function is the library initialization function necessary for any dynamic libraries. It lets you check if the user is using a compatible version of MXNet. Note that this `version` parameter is passed from MXNet when library is loaded.
+
+            MXReturnValue initialize(int version)
+
+* [supportedOps](./subgraph_lib.cc#L179):
+    * This function provides a copy of the model graph as a JSON string, and provides an interface for identifying which operators should be partitioned into a subgraph. Also this is where a custom partitioner can validate the options specified by the user.
+
+            MXReturnValue supportedOps(
+                std::string json,
+                const int num_ids,
+                int *ids,
+                std::unordered_map<std::string, std::string>& options)
+
+* [REGISTER_PARTITIONER(my_part_name)](./subgraph_lib.cc#L238):
+    * This macro registers the custom partitioner and its properties to MXNet by its name. Notice that a partitioner can have multiple partitioning strategies. This enables multiple *passes* to be run in a single partitioning call from the user. The first argument to `addStrategy` is a user-specified name. The second argument is the `supportedOps` function. The third argument is the name of the subgraph operator to create for each subgraph created during partitioning (see below for more info about subgraph operators). The `setAcceptSubgraph` API registers a callback function that is called for each subgraph created during partitioning (more on this below). Notice that the first argument to this function is the strategy to associate with and the second argument is the `acceptSubgraph` function.
+
+            REGISTER_PARTITIONER(my_part_name)
+            .addStrategy("strategy1", 
+                          supportedOps, 
+                          "_custom_subgraph_op")
+            .setAcceptSubgraph("strategy1", 
+                                acceptSubgraph);
+
+
+Also there are some optional functions you can specify:
+
+* [acceptSubgraph](./subgraph_lib.cc#L220):
+    * This function provides an opportunity to accept/reject a subgraph after MXNet partitions it. It also allows specifying custom attributes on the subgraph (ie. user-generated IDs). If you do not register this function, subgraphs will be accepted by default. 
+
+            MXReturnValue acceptSubgraph(
```

Comment:
The naming of this API seems a little bit strange since it can do both accept and reject a subgraph. Can we just rename it to `MergeSubgraph`?

Reply:
how about `reviewSubgraph`?

Reply:
That sounds better to me.

Reply:
done, changed `reviewSubgraph`